### PR TITLE
Fix env and command replacement

### PR DESCRIPTION
**Pull request by maronuu**
## What
Fix a minor bug when `command` and `env` are replaced.

When replacing `command` and `env` in the template job, a minor bug exists.
When the template job's `command` or `env` is not explicitly set, they are evaluated `None`, which resulted in a value error.

ex:
```bash
 Traceback (most recent call last):
job-quick-starter-drwjj job   File "/usr/src/app/./run_batch.py", line 102, in <module>
job-quick-starter-drwjj job     fire.Fire(main)
job-quick-starter-drwjj job   File "/usr/local/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
job-quick-starter-drwjj job     component_trace = _Fire(component, args, parsed_flag_args, context, name)
job-quick-starter-drwjj job   File "/usr/local/lib/python3.9/site-packages/fire/core.py", line 475, in _Fire
job-quick-starter-drwjj job     component, remaining_args = _CallAndUpdateTrace(
job-quick-starter-drwjj job   File "/usr/local/lib/python3.9/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
job-quick-starter-drwjj job     component = fn(*varargs, **kwargs)
job-quick-starter-drwjj job   File "/usr/src/app/./run_batch.py", line 94, in main
job-quick-starter-drwjj job     Kannon(
job-quick-starter-drwjj job   File "/usr/src/app/./kannon/kannon/master.py", line 73, in build
job-quick-starter-drwjj job     self._exec_bullet_task(task)
job-quick-starter-drwjj job   File "/usr/src/app/./kannon/kannon/master.py", line 113, in _exec_bullet_task
job-quick-starter-drwjj job     job = self._create_child_job_object(
job-quick-starter-drwjj job   File "/usr/src/app/./kannon/kannon/master.py", line 132, in _create_child_job_object
job-quick-starter-drwjj job     assert len(job.spec.template.spec.containers[0].command) == 0, \
job-quick-starter-drwjj job TypeError: object of type 'NoneType' has no len()
```

### CODE REVIEW

**Comment by yokomotod** [comment]
Can you add unit tests within this PR?  `_create_child_job_object()` seems to be simple and pure function so I think it's not hard

File: kannon/master.py [hunk 1]
```diff
@@ -126,16 +126,18 @@ def _create_child_job_object(self, job_name: str, serialized_task: str) -> clien
         ]
         job = deepcopy(self.template_job)
         # replace command
-        assert len(job.spec.template.spec.containers[0].command) == 0, \
-            "command will be replaced by kannon, so you shouldn't set any command and args"
+        original_command = job.spec.template.spec.containers[0].command
+        if original_command:
```
Comment (yokomotod) [inline kannon/master.py:130]: unnecessary temporary variable `original_command` ?

Comment (kitagry) [reply kannon/master.py:130]: I think it is better

```python
 assert lob.spec.template.spec.containers[0].command is None, \
            "command will be replaced by kannon, so you shouldn't set any command and args"
```

File: kannon/master.py [hunk 2]
```diff
@@ -126,16 +126,18 @@ def _create_child_job_object(self, job_name: str, serialized_task: str) -> clien
...
+            raise ValueError("command will be replaced by kannon, so you shouldn't set any command and args")
         job.spec.template.spec.containers[0].command = cmd
         # replace env
-        child_envs = []
+        original_env = job.spec.template.spec.containers[0].env
+        child_envs = [] if original_env is None else original_env
```
[the remark this answers is not in the document]
Comment (maronuu) [reply kannon/master.py:135]: I'd prefer the first one!

**Comment by maronuu** [comment]
@yokomotod I've added test cases

**Review by yokomotod — APPROVED** [review]
@maronuu thanks to add tests! lgtm.

but before merge this PR, please merge or rebase the master branch in order to import #13

**Comment by yokomotod** [comment]
> but before merge this PR, please merge or rebase the master branch in order to import !13

@maronuu @kitagry  I enabled the `Require branches to be up to date before merging` setting 


image

so now we really can not merge

image

**Comment by maronuu** [comment]
> I enabled the Require branches to be up to date before merging setting

Thank you!

**Pull request merged**